test/less/functions-each.less

```diff
@@ -0,0 +1,68 @@
+@selectors: blue, green, red;
+@list: a b c d;
+
+each(@selectors, {
+  .sel-@{value} {
+    a: b;
+  }
+});
+
+.each {
+  each(@list, {
+    index+: @index;
+    item@{index}: @value;
+  });
+
+  // nested each
+  each(10px 15px, 20px 25px; { 
+    @alpha: a b c d;
+    @i: (@index - 1);
+
+    each(@value; {
+      @a: extract(@alpha, (@i * 2 + @key));
```

Comment:
I'm having trouble following this test.  Is it to test calculations with index and key?  Maybe we split it into 2 tests, one to test loops in loops, and another to test calculations?  Or we could just add comments to clarify what's being tested.

Reply:
Yeah, that actually threw me when I went back and saw it, even though I wrote it (a couple years ago). Basically it's just demoing that you can get a reference the scoped index of parent in a nested loop. So in an loop of a loop, you can get parent values / indexes along with local by assigning them to a different var in the parent.

Reply:
Hmmm... 🤔 this won't actually work the way I had it before, because of lazy evaluation. If you do `@i: @index` for example, it's evaluated when called.

Reply:
So I'm wondering what to do about this. Technically you can nest `each()`, but if you can't specify your var names, and vars are lazy evaluated, then you can't reference the "parent" index/key/value. Does that matter?

Reply:
Maybe you need to go the generic route and add a function that can be used to selectively force in-place/eager evaluation?

Reply:
> Maybe you need to go the generic route and add a function that can be used to selectively force in-place/eager evaluation? Would maybe help with the &() feature as well, since that also needs to be eager.

Can you clarify? Note that I wrote this comment before defining / implementing the anonymous mixin syntax. I really don't think we want to change the behavior of scoped vars.

Reply:
Just something that I was musing about.

There is - in a way - a method already to force eager evaluation of a variable; it has to be done before offering the variable's value to the function in a function call node. The 'problem' there is that the evaluation of the actual _function_ to which the variable is passed is still lazy.

But if you could force _that_ to be eager and just cache its value until pulled from, then you would be able to force-eval loop variables at the correct time and possibly support other interesting / complex scenarios that have problems with lazy evaluation right now.

Ofcourse; with custom parameter names, these problems should be resolved as there is no lazy chain of variable re-assignments of a value to walk through. So the point - atleast for this PR - no longer applies.

Reply:
Okay, yeah for this one, it's no longer an issue once you can name vars.

Do you approve of this PR as written then?

Reply:
@calvinjuarez Also it would be good to get your sign-off if good too, since it's a new syntax construct for Less. If there's hesitation about it, let me know.